zq crashes when the zql char count exceeds 255 chars

I built a longish search that looked like `uid=foo or uid=bar or ... or uid=baz`. The search caused zq to crash. The crash occurs once the zql character count exceeds 255 characters.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x132d84b]

goroutine 1 [running]:
main.fileExists(0x7ffeefbff473, 0x100, 0xc0000000d4)
        zq/cmd/zq/zq.go:172 +0xab
main.(*Command).Run(0xc0000ea120, 0xc000020070, 0x2, 0x2, 0x0, 0x0)
        zq/cmd/zq/zq.go:216 +0x172
github.com/mccanne/charm.(*instance).run(0xc00000e6a0, 0xc000020070, 0x2, 0x2, 0x0, 0x0)
        go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:53 +0x28c
github.com/mccanne/charm.(*Spec).ExecRoot(0x16fef20, 0xc000020070, 0x2, 0x2, 0xc000115f50, 0x100741f, 0xc0000a6058, 0x0)
        go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:77 +0x96
main.main()
        zq/cmd/zq/main.go:9 +0x78
```

The trace above and the code snippet below is for zq v0.8.0.

```go
164 func fileExists(path string) bool {
165     if path == "-" {
166         return true
167     }
168     info, err := os.Stat(path)
169     if os.IsNotExist(err) {
170         return false
171     }
172     return !info.IsDir()
173 }
```

The problem is `fileExists()` isn't handling the case of `err` on L168 being anything other than "file does not exist". In my case I'm getting an effective `ENAMETOOLONG`, at which point L169 is false and the crash happens on L172.

--- a/cmd/zq/zq.go
+++ b/cmd/zq/zq.go
@@ -123,7 +123,7 @@ func fileExists(path string) bool {
 		return true
 	}
 	info, err := os.Stat(path)
-	if os.IsNotExist(err) {
+	if err != nil {
 		return false
 	}
 	return !info.IsDir()

